roles/ceph-osd/tasks/check_devices.yml

```diff
@@ -16,6 +16,18 @@
   changed_when: false
   failed_when: false
   register: ispartition
+  when: not osd_auto_discovery
+
+- name: check if the device is a partition (autodiscover disks)
+  shell: "echo '/dev/{{ item.key }}' | egrep '/dev/(sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}$'"
```

Comment:
Could the regexp be simplified slightly from:
``'/dev/(sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}$'`` to:
``'/dev/({h,s,v}d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}$'`` ?

The 'v' would cause it to also look for virtualized disks.